src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala

```diff
@@ -490,4 +492,51 @@ final case class AgreementsApiServiceImpl(
     }
   }
 
+  override def getAgreementEServiceProducers(q: Option[String], offset: Int, limit: Int)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerCompactAgreementEServices: ToEntityMarshaller[CompactEServicesLight],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    logger.info(
+      s"Retrieving producer eservices from agreement filtered by eservice name $q, offset $offset, limit $limit"
+    )
+
+    val result: Future[CompactEServicesLight] = for {
+      _            <- validateQueryName(q)
+      requesterId  <- getOrganizationIdFutureUUID(contexts)
+      pagedResults <- agreementProcessService.getAgreementEServices(
+        eServiceName = q,
+        producersIds = Seq(requesterId),
+        consumersIds = Seq.empty,
+        limit = limit,
+        offset = offset
+      )
+    } yield CompactEServicesLight(
+      results = pagedResults.results.map(t => CompactEServiceLight(id = t.id, name = t.name)),
+      pagination = Pagination(offset = offset, limit = limit, totalCount = pagedResults.totalCount)
+    )
+
+    onComplete(result) {
+      handleError(
+        s"Error retrieving eservices from agreement filtered by eservice name $q, offset $offset, limit $limit"
+      ) orElse {
+        case Failure(_: InvalidQueryParameter) =>
+          getAgreementEServiceProducers200(
+            CompactEServicesLight(
+              results = Nil,
+              pagination = Pagination(offset = offset, limit = limit, totalCount = 0)
+            )
+          )
+        case Success(resource)                 =>
+          getAgreementEServiceProducers200(resource)
+      }
+    }
+  }
+
+  private def validateQueryName(q: Option[String]): Future[Unit] = {
+    q match {
+      case Some(value) if value.length < 3 => Future.failed(InvalidQueryParameter(value))
+      case _                               => Future.successful(())
+    }
+  }
```

Comment:
[ditto](https://github.com/pagopa/interop-be-backend-for-frontend/pull/105#discussion_r1108706548)